nautobot/dcim/models/device_components.py

```diff
@@ -773,6 +778,87 @@ def parent(self):
         return self.device
 
 
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "statuses",
+    "webhooks",
+)
+class InterfaceRedundancyGroup(PrimaryModel, ConfigContextModel, StatusModel):  # pylint: disable=too-many-ancestors
+    """
+    A collection of Interfaces that supply a redundancy group for protocols like HSRP/VRRP.
+    """
+
+    name = models.CharField(max_length=100, unique=True)
+    slug = AutoSlugField(populate_from="name")
+    description = models.CharField(max_length=200, blank=True)
+    virtual_ip = models.ForeignKey(
+        to="ipam.IPAddress",
+        on_delete=models.CASCADE,
+        null=True,
+        related_name="interface_redundancy_virtual_ip",
```

Comment:
Since this is a 1-to-many, this related name should be 1.) plural, 2.) reference the model from which the relation is sourced; so `interface_redundancy_groups` in this case.

Reply:
much better name (DOH!)